src/python/foglamp/core/server.py

```diff
@@ -40,15 +55,117 @@ def _make_app():
         routes.setup(app)
         return app
 
+    @staticmethod
+    def _make_core_app():
+        """Creates the Service management REST server Core a.k.a. service registry
+
+        :rtype: web.Application
+        """
+        app = web.Application(middlewares=[middleware.error_middleware])
+        routes.core_setup(app)
+        return app
+
     @classmethod
     async def _start_scheduler(cls):
         """Starts the scheduler"""
-        cls.scheduler = Scheduler()
+        _logger.info("start scheduler")
+        cls.scheduler = Scheduler(cls.core_management_port)
         await cls.scheduler.start()
 
+    @staticmethod
+    def __start_storage(host, m_port):
+        _logger.info("start storage, from directory %s", _STORAGE_DIR)
+        try:
+            cmd_with_args = ['./storage', '--address={}'.format(host),
+                             '--port={}'.format(m_port)]
+            subprocess.call(cmd_with_args, cwd=_STORAGE_DIR)
+        except Exception as ex:
+            _logger.exception(str(ex))
+
+    @classmethod
+    async def _start_storage(cls, loop):
+        if loop is None:
+            loop = asyncio.get_event_loop()
+            # callback with args
+        loop.call_soon(cls.__start_storage, cls._host, cls.core_management_port)
+
+    @classmethod
+    def _start_app(cls, loop, app, host, port):
+        if loop is None:
+            loop = asyncio.get_event_loop()
+
+        handler = app.make_handler()
+        coro = loop.create_server(handler, host, port)
+        server = loop.run_until_complete(coro)
+        return server, handler
+
+    @classmethod
+    def _start_core(cls, loop=None):
+        _logger.info("start core")
+
+        try:
+            host = cls._host
+
+            core_app = cls._make_core_app()
+            core_server, core_server_handler = cls._start_app(loop, core_app, host, 0)
+            address, cls.core_management_port = core_server.sockets[0].getsockname()
+            _logger.info('Management API started on http://%s:%s', address, cls.core_management_port)
+            # make sure to
+            loop.run_until_complete(core_app.startup())
+
+            service_app = cls._make_app()
+            service_server, service_server_handler = cls._start_app(loop, service_app, host, cls.rest_service_port)
+            address, service_server_port = service_server.sockets[0].getsockname()
+            _logger.info('Rest Server started on http://%s:%s', address, service_server_port)
+
+            # register now
+            cls._register_core(host, cls.core_management_port, service_server_port)
+            print("(Press CTRL+C to quit)")
+            try:
+                # start storage
+                loop.run_until_complete(cls._start_storage(loop))
```

Comment:
with this is place we're in a much better position to start the storage service within the scheduler

Reply:
you mean as a schedule? please see that way in PR 224